### DISCUSSION TITLE
Phoenix integration: WebRTC video component

### DISCUSSION THREAD

**Discussion by zachallaun** [elixir-webrtc/ex_webrtc#164] [Ideas]
Hey all -- really cool project here!

I was wondering if there were any plans to package the client-side setup into a built-in video component and hook. For instance, the Broadcaster app uses [this custom hook](https://github.com/elixir-webrtc/apps/blob/master/broadcaster/assets/js/home.js) and I wonder if an `ex_webrtc_phoenix` could be published that basically packages all that up so that someone could build a basic WebRTC integration while writing essentially no JavaScript.

While I admit to having not dug too deeply into the code, it looks like most of the assumed variables in the hook (like the ICE servers, WHEP endpoint, etc.) could be passed as attributes.

This would have the side-benefit of potentially making the Broadcaster app example a bit slicker as the JS would be abstracted away, and LiveView streams could even be used for the chat instead of channels + DOM mutation 🙂

**Comment by LVala** [comment]
That's a cool idea (we thought of something similar at some point too, albeit its unrelated to the WebRTC implementation itself, so it landed in a drawer) - it might be worth tackling @mickel8

**Reply by zachallaun** [reply]
Yeah, I don't think it's something that belongs in `ex_webrtc`, but an `ex_webrtc_phoenix` would fit the bill!

**Comment by mickel8** [comment]
That would be awesome! Personally, I would love someone who knows LiveView better to implement this. I can help with explaining WebRTC. If not, I will try to create some PoC :)

**Reply by zachallaun** [reply]
I unfortunately don't have the bandwidth to commit to that, but if I find some time to play around, I'll drop a note here and share anything I come up with!

**Reply by mickel8** [reply]
Sure! Even references or examples that we could take a look at would be helpful. Just to know where to start

**Reply by zachallaun** [reply]
Very generally, [LiveToast](https://github.com/srcrip/live_toast) is a small, really well-designed library that replaces the default Phoenix Flash system. It's a good example for how to package components and hooks for reuse.

**Reply by mickel8** [reply]
Perfect! Thank you, that's all I need :)

**Comment by mickel8** [comment]
Hi @zachallaun, JFYI, check out: https://github.com/elixir-webrtc/live_ex_webrtc

**Reply by zachallaun** [reply]
Awesome! I'll give this a try tomorrow!

**Reply by mickel8** [reply]
See also this video: https://www.youtube.com/watch?v=jziOb2Edfzk&t=991s
